packages/roosterjs-editor-dom/lib/userPosition/adjustUserPosition.ts

```diff
@@ -0,0 +1,238 @@
+import {
+    EditorCore,
+    NodePosition,
+    NodeType,
+    PositionType,
+    QueryScope,
+} from 'roosterjs-editor-types';
+import {
+    changeElementTag,
+    contains,
+    createRange,
+    findClosestElementAncestor,
+    getBlockElementAtNode,
+    getTagOfNode,
+    isNodeEmpty,
+    isPositionAtBeginningOf,
+    isVoidHtmlElement,
+    LinkInlineElement,
+    Position,
+    PositionContentSearcher,
+    queryElements,
+    splitBalancedNodeRange,
+    splitTextNode,
+    toArray,
+    unwrap,
+    VTable,
+    wrap,
+} from 'roosterjs-editor-dom';
+
+/**
+ * Adjust position for <a> tag don't be nested inside another <a> tag.
+ */
+
+export function adjustInsertPositionForHyperLink(
+    root: HTMLElement,
+    nodeToInsert: Node,
+    position: NodePosition,
```

Comment:
let's accept just Position interface. We only use "NodePosition" when we want to create new instance.